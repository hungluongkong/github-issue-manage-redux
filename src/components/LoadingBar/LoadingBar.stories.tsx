import React from 'react';
import LoadingBar from '.';
import { storiesOf } from '@storybook/react';
import { PreviewArea, Code } from '../commonStyled';

storiesOf('Loading Bar', module)
  .add('Default', (): JSX.Element => (
    <>
      <h3>Loading Bar</h3>
      <PreviewArea>
        <LoadingBar />
      </PreviewArea>
      <Code>
        {`
          <LoadingBar />
        `}
      </Code>
    </>
  ));
