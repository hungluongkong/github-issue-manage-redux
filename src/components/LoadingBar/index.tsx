import * as React from 'react';

import styled, { keyframes } from 'styled-components';
import * as color from '../../theme/color';

const loadingKeyFrame = keyframes`
  0% {
    top: 8px;
    height: 64px;
  }
  50%, 100% {
    top: 24px;
    height: 32px;
  }
`;

const LoadingWrapper = styled.div`
  text-align: center;
`;

const LoadBar = styled.div`
  display: inline-block;
  position: relative;
  width: 80px;
  height: 80px;
`;

const LoadingBlock = styled.div`
  display: inline-block;
  position: absolute;
  left: 8px;
  width: 16px;
  background: ${color.LOADING_BAR_COLOR};
  animation: ${loadingKeyFrame} 1.2s cubic-bezier(0, 0.5, 0.5, 1) infinite;

  :nth-child(1) {
    left: 8px;
    animation-delay: -0.24s;
  }
  :nth-child(2) {
    left: 32px;
    animation-delay: -0.12s;
  }
  :nth-child(3) {
    left: 56px;
    animation-delay: 0;
  }
`;

const LoadingBar = (): JSX.Element => (
  <LoadingWrapper>
    <LoadBar>
      {/* Three loading box effect */}
      <LoadingBlock />
      <LoadingBlock />
      <LoadingBlock />
    </LoadBar>
  </LoadingWrapper>
);

export default LoadingBar;
