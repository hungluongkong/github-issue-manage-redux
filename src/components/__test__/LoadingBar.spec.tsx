import React from 'react';
import { create } from 'react-test-renderer';
import LoadingBar from '../LoadingBar';

describe('Unit test for LoadingBar component', () => {
  it('snapshot', () => {
    const loadingBar = create(<LoadingBar />);
    expect(loadingBar).toMatchSnapshot();
  });
});
