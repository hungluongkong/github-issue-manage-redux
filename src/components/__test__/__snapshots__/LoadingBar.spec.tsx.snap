// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for LoadingBar component snapshot 1`] = `
<div
  className="sc-AxjAm jWlVxK"
>
  <div
    className="sc-AxirZ ghRTLL"
  >
    <div
      className="sc-AxiKw eoswRd"
    />
    <div
      className="sc-AxiKw eoswRd"
    />
    <div
      className="sc-AxiKw eoswRd"
    />
  </div>
</div>
`;
